portability-core/src/main/java/org/dataportabilityproject/cloud/google/ProjectId.java

```diff
@@ -0,0 +1,24 @@
+package org.dataportabilityproject.cloud.google;
+
+import com.google.inject.Inject;
+
+/**
+ * Wrapper class for a Google Cloud Platform project ID.
+ *
+ * <p>This class exists so that we can inject ProjectId into Google implementation classes without
+ * forcing Guice to eagerly eval Provider<GoogleCloudFactory> when not using cloud=GOOGLE. i.e. if
+ * we just bound the project ID as a String directly ("@ProjectId String projectId") via
+ * bind(String.toClass).annotatedWith(ProjectId.class).toInstance(...), that causes eager init.
+ */
+class ProjectId {
```

Comment:
Could we use Guice bind(String.class).annotatedWith(ProjectId.class).to("the env var") to avoid having a wrapper class?

Reply:
Done, see other reply